File: python/src/epostak/resources/enterprise.py
"""Workflow-first Enterprise API namespace."""

from __future__ import annotations

from typing import Any


class EnterprisePullResource:
    """Cursor-based Enterprise pull APIs."""

    def __init__(self, inbound: Any, outbound: Any) -> None:
        self.inbound = inbound
        self.outbound = outbound


class EnterpriseResource:
    """Groups all `/api/v1/*` Enterprise resources under one namespace."""

    def __init__(self, client: Any) -> None:
        self.auth = client.auth
        self.audit = client.audit
        self.documents = client.documents
        self.inbox = client.documents.inbox
        self.firms = client.firms
        self.peppol = client.peppol
        self.webhooks = client.webhooks
        self.reporting = client.reporting
        self.extract = client.extract
        self.account = client.account
        self.integrator = client.integrator
        self.connector = client.connector
        self.pull = EnterprisePullResource(client.inbound, client.outbound)

File: ruby/lib/epostak/resources/enterprise.rb
# frozen_string_literal: true

module EPostak
  module Resources
    class EnterprisePull
      attr_reader :inbound, :outbound

      def initialize(inbound:, outbound:)
        @inbound = inbound
        @outbound = outbound
      end
    end

    class Enterprise
      attr_reader :auth, :audit, :documents, :inbox, :firms, :peppol,
                  :webhooks, :reporting, :account, :extract, :integrator,
                  :connector, :payloads, :events, :box, :pull

      def initialize(client)
        @auth = client.auth
        @box = client.box
        @audit = client.audit
        @documents = client.documents
        @inbox = client.documents.inbox
        @firms = client.firms
        @peppol = client.peppol
        @webhooks = client.webhooks
        @reporting = client.reporting
        @account = client.account
        @extract = client.extract
        @integrator = client.integrator
        @connector = client.connector
        @payloads = client.payloads
        @events = client.events
        @pull = EnterprisePull.new(inbound: client.inbound, outbound: client.outbound)
      end
    end
  end
end
